### ISSUE TITLE
[Bug]: can't use `swc-plugin-coverage-instrument` with rsbuild

### ISSUE THREAD

**Issue by eidellev** [web-infra-dev/rsbuild#2036]
### Version

```sh
System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M2 Pro
    Memory: 8.07 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Brave Browser: 114.1.52.130
    Chrome: 123.0.6312.107
    Safari: 17.2.1
  npmPackages:
    @rsbuild/core: ^0.5.9 => 0.5.9 
    @rsbuild/plugin-react: ^0.5.9 => 0.5.9
```

### Details

Hi

I am migrating my `webpack` app to `rspack`.
Really impressed so far with the speed and overall developer experience. Keep up the great work!

I'm trying to use `swc-plugin-coverage-instrument` but getting the following error:
```
Html Webpack Plugin:

Error: [html-rspack-plugin]: [html-rspack-plugin]:   × Module parse failed:
  ╰─▶   × JavaScript parsing error: The left-hand side of an assignment expression must be a variable or a property access.
          ╭─[32:1]
       32 │     if (!coverage[path] || coverage[path].hash !== hash) this = this;
       33 │     var actualCoverage = coverage[path];
       34 │     this = this;
          ·     ────
       35 │     return actualCoverage;
       36 │ }
          ╰────
      
  help: 
        File was processed with these loaders:
         * builtin:swc-loader??ruleSet[1].rules[3].use[0]
        You may need an additional loader to handle the result of these loaders.
```

My `rspack` config:
```
import { defineConfig } from '@rsbuild/core';
import { pluginReact } from '@rsbuild/plugin-react';

export default defineConfig({
  plugins: [pluginReact()],
  tools: {
    swc: {
      jsc: {
        experimental: {
          plugins: [['swc-plugin-coverage-instrument', {}]],
        },
      },
    },
  },
});

```

Any help would be welcome :-)

### Reproduce link

https://github.com/eidellev/rspack-bug-reproduction

### Reproduce Steps

```
npm run dev
```

**Comment by SoonIter** [comment]
It may be related to https://github.com/web-infra-dev/rspack/issues/5921. 
The version of swc-plugin must be aligned with the rspack builtin swc_core .

**Comment by eidellev** [comment]
Thank you @SoonIter for the quick reply.

I don't think this is related. I tried rolling back the plugin to an older version and got a panic like in the issue you linked to.

**Comment by xc2** [comment]
rspack 0.6.0 which should fix this issue will be released soon.

please wait for the new version of rsbuild that comes with rspack 0.6.0 or try out @rspack/core@nightly

**Comment by eidellev** [comment]
@xc2 Thank you!

**Comment by chenjiahan** [comment]
Fixed in v0.6.0

**Comment by eidellev** [comment]
Works great. Thanks @chenjiahan !
